app/src/main/java/com/io/moviO/domain/GetMoviesUseCase.kt

```diff
@@ -1,11 +1,18 @@
 package com.io.moviO.domain
 
+import com.io.moviO.data.DataResult
 import com.io.moviO.data.Movie
 import com.io.moviO.data.MoviesRepository
 
-class GetMoviesUseCase {
+class GetMoviesUseCase : UseCase<MoviesRepository, DataResult<List<Movie>>> {
     private val moviesRepo = MoviesRepository
 
-    fun getMovies(): List<Movie> = moviesRepo.getMovies()
-
+    override suspend fun execute(param: MoviesRepository): DataResult<List<Movie>> {
```

Comment:
MoviesRepository is not the input for getting a list of movies. An "input" for a UseCase would be the id to get the movie for. In that UseCase you have `UseCase<String, DataResult<Movie>>` but not the Repo as input. Why does this one have it then?

This UseCase just doesn't have an imput for it's starting point. :)

Reply:
resolved